src/coreclr/vm/jithelpers.cpp

```diff
@@ -1685,25 +1685,35 @@ HCIMPL1(void, IL_ThrowExact, Object* obj)
 
     FC_GC_POLL_NOT_NEEDED();    // throws always open up for GC
 
-    HELPER_METHOD_FRAME_BEGIN_ATTRIB_NOPOLL(Frame::FRAME_ATTR_EXCEPTION);    // Set up a frame
-
     OBJECTREF oref = ObjectToOBJECTREF(obj);
+    GetThread()->GetExceptionState()->SetRaisingForeignException();
+
+#ifdef FEATURE_EH_FUNCLETS
+    Thread *pThread = GetThread();
+    
+    SoftwareExceptionFrame exceptionFrame;
+#ifdef TARGET_X86
+    _ASSERTE(!"Introduce `IL_ThrowExact_x86` following the pattern of other Throw helpers.");
```

Comment:
Does it mean the EH test fails on x86 now? Can we disable it somehow?

Reply:
x86 does not do `FEATURE_EH_FUNCLETS` right now so it will work. Thus we can postpone introducing `IL_ThrowExact_x86`.

When x86 is moved onto `FEATURE_EH_FUNCLETS` plan, which I understand will happen at some point, we can remove `HELPER_METHOD_FRAME_BEGIN_ATTRIB_NOPOLL` path, which is desirable, but will also need to implement `IL_ThrowExact_x86`. It takes an extra parameter, I have only vague idea why.

Reply:
All async tests should be passing right now - that is on win-x64 and win-x86, where it is easy to test once in a while.

Reply:
> When x86 is moved onto FEATURE_EH_FUNCLETS plan, which I understand will happen at some point, 

It is expected to happen very soon. The changes are in https://github.com/dotnet/runtime/pull/113576. The last remaining item before this gets enabled is private diagnostic tests run. If you would like to get ahead of this, you can apply the 10 line change locally and do the required fixes to make things work.

Reply:
Context about last testing item: https://github.com/dotnet/runtime/issues/113985#issuecomment-2798207726.

Reply:
> It takes an extra parameter, I have only vague idea why.

win-x86 doesn't have unwinding information for native code. Thus we need to capture the precise register context in assembly code before giving control to the C code.

This should be fairly mechanical change so I can make a PR for the code.

Reply:
https://github.com/dotnet/runtimelab/pull/3100